lib/maths/CBoostedTreeImpl.cc

```diff
@@ -477,11 +477,14 @@ CBoostedTreeImpl& CBoostedTreeImpl::operator=(CBoostedTreeImpl&&) = default;
 
 void CBoostedTreeImpl::train(core::CDataFrame& frame,
                              const TTrainingStateCallback& recordTrainStateCallback) {
-    auto recordProgress{std::function<void(double)>()};
-    auto recordMemoryUsage{std::function<void(std::int64_t)>()};
+    std::function<void(double)> recordProgress;
+    std::function<void(std::int64_t)> recordMemoryUsage;
```

Comment:
nit: can we use typedefs for these: we already have them and if we change the type in the future it saves having to fix this code.

Reply:
Fixed in ec7324ed